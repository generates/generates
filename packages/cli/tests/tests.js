const { test } = require('@ianwalter/bff')
const execa = require('execa')
const exampleCli = require('./fixtures/exampleCli')

test('cli', async t => {
  const { stdout } = await execa('./tests/fixtures/exampleCli.js', ['-c', '4'])
  t.expect(stdout).toMatchSnapshot()
})

test('dot format', async t => {
  const args = ['--some.other.thing', 'b']
  const { stdout } = await execa('./tests/fixtures/exampleCli.js', args)
  t.expect(stdout).toMatchSnapshot()
})

test('default', async t => {
  t.expect(exampleCli.path).toBe('/some/path')
})
