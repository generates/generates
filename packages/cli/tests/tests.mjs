import { test } from '@ianwalter/bff'
import execa from 'execa'
import exampleCli from './fixtures/exampleCli.js'

test('cli', async t => {
  const { stdout } = await execa('./tests/fixtures/exampleCli.js', ['-c', '4'])
  t.expect(stdout).toMatchSnapshot()
})

test('dot format', async t => {
  const args = ['--some.other.thing', 'b']
  const { stdout } = await execa('./tests/fixtures/exampleCli.js', args)
  t.expect(stdout).toMatchSnapshot()
})

test('default', async t => {
  t.expect(exampleCli.path).toBe('/some/path')
})

test('command', async t => {
  let args = ['save', '--path', './test.mp3']
  let result = await execa('./tests/fixtures/musicCli.js', args)
  t.expect(result.stdout).toContain('Music saved to:', args[2])

  result = await execa('./tests/fixtures/musicCli.js', ['start', 'drum'])
  t.expect(result.stdout).toContain('No tempo specified')

  args = ['start', 'drum', '120']
  result = await execa('./tests/fixtures/musicCli.js', args)
  t.expect(result.stdout).toContain('Playing the drum at', args[2])

  result = await execa('./tests/fixtures/musicCli.js', ['start', 'guitar'])
  t.expect(result.stdout).toContain('Playing guitar!')
})

test('command help', async t => {
  let result = await execa('./tests/fixtures/musicCli.js', ['--help'])
  t.expect(result.stdout).toMatchSnapshot()

  result = await execa('./tests/fixtures/musicCli.js', ['start', '--help'])
  t.expect(result.stdout).toMatchSnapshot()

  const args = ['start', 'drum', '--help']
  result = await execa('./tests/fixtures/musicCli.js', args)
  t.expect(result.stdout).toMatchSnapshot()

  result = await execa('./tests/fixtures/musicCli.js', ['play', 'drum'])
  t.expect(result.stdout).toMatchSnapshot()
})
