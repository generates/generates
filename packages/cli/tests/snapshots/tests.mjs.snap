// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli 1`] = `
"     1 {
     2   level: 'debug',
     3   processorConcurrency: 4,
     4   some: {
     5     thing: true,
     6     other: {
     7       thing: 'a'
     8     }
     9   },
    10   packageJson: {
    11     name: '@generates/cli'
    12   },
    13   c: 4,
    14   path: '/some/path',
    15   debug: false,
    16   d: false,
    17   args: []
    18 }"
`;

exports[`command help 1`] = `
"
💁  [36m[1m# music[22m[39m

    Compose music from the command line!

    ## Usage

    music [commands] [args] [options]

    ## Commands

      * start, play – Start playing instruments
      * save – Save the music to a file
"
`;

exports[`command help 2`] = `
"
💁  [36m[1m# music ❯ start[22m[39m

    Start playing instruments

    ## Usage

    music start [instrument] [args]

    ## Commands

      * drum – Start playing the drum at a certain tempo
      * guitar – Start playing the guitar
"
`;

exports[`command help 3`] = `
"
💁  [36m[1m# music ❯ start ❯ drum[22m[39m

    Start playing the drum at a certain tempo

    ## Usage

    music start drum [tempo]
"
`;

exports[`command help 4`] = `
"
⚠️   [33m[1mCommand \\"begin\\" not found[22m[39m

💁  [36m[1m# music[22m[39m

    Compose music from the command line!

    ## Usage

    music [commands] [args] [options]

    ## Commands

      * start, play – Start playing instruments
      * save – Save the music to a file
"
`;

exports[`dot format 1`] = `
"     1 {
     2   level: 'debug',
     3   processorConcurrency: 1,
     4   some: {
     5     thing: true,
     6     other: {
     7       thing: 'b'
     8     }
     9   },
    10   packageJson: {
    11     name: '@generates/cli'
    12   },
    13   c: 1,
    14   path: '/some/path',
    15   debug: false,
    16   d: false,
    17   args: []
    18 }"
`;
